jspwiki-main/src/main/java/org/apache/wiki/servlets/WikiServletContextListener.java

```diff
@@ -0,0 +1,87 @@
+/*
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+ */
+package org.apache.wiki.servlets;
+
+import org.apache.wiki.InternalWikiException;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import java.io.File;
+
+/**
+ * A {@link ServletContextListener} implementation which sets up the necessary configurations
+ * for the application at servlet context initialization time. Specifically, it retrieves
+ * the servlet's default temporary directory and stores its absolute path in the
+ * servlet context as "wikiTempDir". This stored path is then utilized by
+ * other components of the application (such as WikiEngine) as the default
+ * working directory.
+ *
+ * <p>Through this implementation, the application ensures that the default working
+ * directory is the servlet container's temporary directory for each context, instead of
+ * the system temporary directory, thus enabling a more robust and secure configuration.
+ *
+ * @since 2.12.0
+ */
+public class WikiServletContextListener implements ServletContextListener {
```

Comment:
Hi @arturobernalg , thanks for looking into this!

Instead of adding a new servlet, I think it would be better to go through `WikiBootstrapServletContextListener`, which  calls [`Wiki.init( servletContext)`](https://github.com/apache/jspwiki/blob/master/jspwiki-bootstrap/src/main/java/org/apache/wiki/bootstrap/WikiBootstrapServletContextListener.java#L57), which in turn delegates to [`PropertyReader.loadWebAppProps( servletContext )`](https://github.com/apache/jspwiki/blob/master/jspwiki-util/src/main/java/org/apache/wiki/util/PropertyReader.java#L121) to load the wiki properties (from `jspwiki[-custom].properties`, system env & props, etc.). I'd check there for `jspwiki.workDir` existence. If it's set, then we do nothing. If it's missing, then we set it to `javax.servlet.context.tempdir` (or, if this doesn't exist, `java.io.tmpdir`) on the properties object. 

Once the `jspwiki.workDir` is set on `PropertyReader`, then we can [remove the `java.io.tmpdir` usage on `WikiEngine`](https://github.com/apache/jspwiki/blob/master/jspwiki-main/src/main/java/org/apache/wiki/WikiEngine.java#L351-L353), and refactor [`Installer`](https://github.com/apache/jspwiki/blob/master/jspwiki-main/src/main/java/org/apache/wiki/ui/Installer.java#L69) to use the wiki properties instead of directly using `java.io.tmpdir`.

WDYT?

Reply:
Hi @juanpablo-santos 

I've made the changes as per your suggestion. I modified WikiBootstrapServletContextListener to check and set 'jspwiki.workDir' and refactored Installer to use this property instead of 'java.io.tmpdir'. I also removed the direct usage of 'java.io.tmpdir' in WikiEngine. Please have a look at the latest commit.

Thanks for your valuable feedback!